File: app/src/main/java/ru/vezdekod/podcast/fragments/StartFragment.java
package ru.vezdekod.podcast.fragments;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavDirections;

import ru.vezdekod.podcast.OnFragmentInteractionListener;
import ru.vezdekod.podcast.R;

public class StartFragment extends Fragment {

    private OnFragmentInteractionListener onFragmentInteractionListener;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        onFragmentInteractionListener = (OnFragmentInteractionListener) context;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_start, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        view.findViewById(R.id.fragment_start___button___create_podcast).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                NavDirections navDirections = StartFragmentDirections.actionNavFirstToNavMainPodcastData();
                onFragmentInteractionListener.onFragmentInteraction(navDirections);
            }
        });
    }
}

File: app/src/main/java/ru/vezdekod/podcast/fragments/StartFragment.kt
package ru.vezdekod.podcast.fragments

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.navigation.NavDirections
import ru.vezdekod.podcast.OnFragmentInteractionListener
import ru.vezdekod.podcast.databinding.FragmentStartBinding

class StartFragment : Fragment() {

    private lateinit var viewBinding: FragmentStartBinding
    private var onFragmentInteractionListener: OnFragmentInteractionListener? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        onFragmentInteractionListener = context as OnFragmentInteractionListener
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View = FragmentStartBinding.inflate(inflater, container, false).also {
        viewBinding = it
    }.root

    override fun onResume() {
        super.onResume()
        (requireActivity() as AppCompatActivity).supportActionBar?.hide()
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewBinding.closeMessageButton.setOnClickListener {
            Toast.makeText(context, "Заглушка", Toast.LENGTH_SHORT).show()
        }
        viewBinding.createPodcastButton.setOnClickListener {
            val navDirections: NavDirections =
                StartFragmentDirections.actionNavFirstToNavMainPodcastData()
            onFragmentInteractionListener?.onFragmentInteraction(navDirections)
        }
    }
}